Library/ViewModels/ReportProjectFormViewModel.swift

```diff
@@ -5,117 +5,108 @@ import ReactiveSwift
+  @Published public var saveButtonEnabled: Bool = false
+  @Published public var detailsText: String = ""
+  @Published public var saveTriggered: Bool = false
+  @Published public var bannerMessage: MessageBannerViewViewModel? = nil
 
-  private var cancellables = Set<AnyCancellable>()
+  @Published public var submitSuccess: Bool = false
 
-  public init() {
-    self.saveButtonEnabled = Publishers
-      .CombineLatest(self.projectFlaggingKind, self.detailsText)
-      .map { !$0.1.isEmpty }
-      .eraseToAnyPublisher()
+  private let viewDidLoadSubject = PassthroughSubject<Bool, Never>()
+
+  private var cancellables = Set<AnyCancellable>()
 
-    let userEmailEvent = self.viewDidLoadProperty.signal
-      .switchMap { _ in
+  public let projectID: String
+  public let projectURL: String
+  public let projectFlaggingKind: GraphAPI.FlaggingKind
+
+  public init(projectID: String,
+              projectURL: String,
+              projectFlaggingKind: GraphAPI.FlaggingKind) {
+    self.projectID = projectID
+    self.projectURL = projectURL
+    self.projectFlaggingKind = projectFlaggingKind
+
+    /// Only enable the save button if the user has entered detail text
+    self.$detailsText
+      .map { !$0.isEmpty }
+      .assign(to: &$saveButtonEnabled)
+
+    /// Load the current user's e-mail on page load
+    self.viewDidLoadSubject
+      .flatMap { _ in
         AppEnvironment.current
           .apiService
-          .fetchGraphUser(withStoredCards: false)
-          .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
-          .materialize()
+          .fetchGraphUserEmailCombine()
       }
-
-    self.userEmail = userEmailEvent.values().map { $0.me.email ?? "" }
-
-    _ = self.userEmail
-      .observeForUI()
-      .observeValues { [weak self] email in
-        self?.retrievedEmail.send(email)
+      .compactMap { envelope in
+        envelope.me.email
       }
+      .catch { _ in
+        CurrentValueSubject("")
+      }
+      .assign(to: &$retrievedEmail)
 
-    let submitReportEvent = self.submitReportProperty.signal.skipNil()
-      .map(CreateFlaggingInput.init(contentId:kind:details:clientMutationId:))
-      .switchMap { input in
-        AppEnvironment
-          .current
-          .apiService
-          .createFlaggingInput(input: input)
-          .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
-          .materialize()
+    /// Submits report on saveTriggered
+    self.$saveTriggered
+      .filter { $0 } // only save if saveTriggered is true
+      .compactMap { [weak self] _ in
+        self?.createFlaggingInput()
       }
+      .flatMap { $0 }
+      .sink { [weak self] _ in
+        // An error happens
 
-    _ = submitReportEvent.errors()
-      .observeForUI()
-      .observeValues { [weak self] _ in
-        let messageBannerViewViewModel = MessageBannerViewViewModel((
+        self?.saveTriggered = false
```

Comment:
Some of this logic was in the view; I moved into the view model. These are hard to read in the git diff view so I recommend checking them out in the actual full file view.